src/black/mode.py

```diff
@@ -136,13 +143,23 @@ class Mode:
     experimental_string_processing: bool = False
     preview: bool = False
 
+    def __post_init__(self) -> None:
+        if self.experimental_string_processing:
+            warn(
+                "`experimental string processing` has been included in `preview`"
+                " and deprecated. Use `preview` instead.",
+                Deprecated,
+            )
+
     def __contains__(self, feature: Preview) -> bool:
         """
         Provide `Preview.FEATURE in Mode` syntax that mirrors the ``preview`` flag.
 
         The argument is not checked and features are not differentiated.
         They only exist to make development easier by clarifying intent.
         """
+        if feature is Preview.string_processing:
+            return self.preview or self.experimental_string_processing
         return self.preview
```

Comment:
Looks like mypy (correctly) detects that this line is unreachable. Maybe to satisfy it we can add a feature we know we'll add to `--preview`, such as power operand hugging.

Reply:
I mean just adding the enum definition, not implement the feature in this PR.

Reply:
Yeah, that sounds good!